src/LiveComponent/src/Util/FingerprintCalculator.php

```diff
@@ -20,12 +25,29 @@
  */
 class FingerprintCalculator
 {
-    public function __construct(private string $secret)
-    {
+    public function __construct(
+        private NormalizerInterface $objectNormalizer,
+        private string $secret,
+    ) {
     }
 
     public function calculateFingerprint(array $data): string
     {
-        return base64_encode(hash_hmac('sha256', serialize($data), $this->secret, true));
+        return base64_encode(hash_hmac('sha256', serialize($this->normalizeData($data)), $this->secret, true));
+    }
+
+    private function normalizeData(array $data): array
+    {
+        return array_map(function (mixed $value) {
```

Comment:
Can we just call `return $this->objectNormalizer->normalize($value, context: [LiveComponentHydrator::LIVE_CONTEXT => true]);` (or inline above)?